### PR TITLE
chore: [IA-754] Track migration events

### PR DESCRIPTION
**Pull request by pietro909**
## Short description
Track the migration events (start, success, failure).
An example of the events is reported [here](https://eu.mixpanel.com/project/2460815/view/3003771/app/insights#report/28984148/migration-errors).

### CODE REVIEW

File: ts/store/middlewares/analytics.ts [hunk 1]
```diff
@@ -263,6 +264,22 @@ const trackAction =
         }
         break;
       }
+      case getType(migrateToPaginatedMessages.request): {
+        return mp.track("MESSAGES_MIGRATION_START", {
+          total: action.payload.total
```
Comment (Undermaken) [inline ts/store/middlewares/analytics.ts:269]: > **total**: numero dei messaggi da migrare

Schermata 2022-03-30 alle 12 04 19
As far I know, it seems we are tracing `MessageStatus | undefined` and not a `number`
If it is true, maybe this is not we want

File: ts/store/middlewares/analytics.ts [hunk 2]
```diff
@@ -263,6 +264,22 @@ const trackAction =
...
+      }
+      case getType(migrateToPaginatedMessages.success): {
+        return mp.track("MESSAGES_MIGRATION_SUCCESS", {
+          total: action.payload
+        });
+      }
+      case getType(migrateToPaginatedMessages.failure): {
+        return mp.track("MESSAGES_MIGRATION_FAILURE", {
+          failed: action.payload.failed,
+          succeeded: action.payload.succeeded
```
Comment (Undermaken) [inline ts/store/middlewares/analytics.ts:280]:
```suggestion
          failed: action.payload.failed.length,
          succeeded: action.payload.succeeded.length
```

⚠️ in this way you will trace also the messages id
Schermata 2022-03-30 alle 12 07 22

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: ts/store/middlewares/analytics.ts (hunk 1 above)
Comment (pietro909) [reply ts/store/middlewares/analytics.ts]: Good catch. What we want is the size of that store. Fixed [here](https://github.com/pagopa/io-app/pull/3865/commits/af5257fed09d564aafb42740e28f7fd8009eaa56).

File: ts/store/middlewares/analytics.ts [hunk 3]
```diff
@@ -263,6 +264,22 @@ const trackAction =
         }
         break;
       }
+      case getType(migrateToPaginatedMessages.request): {
+        return mp.track("MESSAGES_MIGRATION_START", {
+          total: Object.keys(action.payload)
```
Comment (Undermaken) [inline ts/store/middlewares/analytics.ts:269]:
```suggestion
          total: Object.keys(action.payload).length
```

**Review by Undermaken — CHANGES_REQUESTED** [review]

**Review by Undermaken — APPROVED** [review]

**Pull request merged**